perf/benchmark/runner/graph.py

```diff
@@ -11,97 +11,121 @@
 import sys
 import argparse 
 
+
 # generate_chart displays numthreads vs. metric, writes to interactive HTML
-def generate_chart(csv, metric):
-    print("Generating chart, metric=%s, csv=%s ..." % (metric, csv))
+def generate_chart(mesh, csv, x_label, y_label_short):
+    print("Generating chart, x_label=%s, y_label=%s, csv=%s ..." % (x_label, y_label_short, csv))
 
     # valid options = latency, memory, cpu
-    valid_metrics = {"p50": "p50", "p90": "p90", "p99": "p99", "mem": "mem_MB_max_fortioserver_deployment_proxy", "cpu": "cpu_mili_max_fortioserver_deployment_proxy"}
+    valid_metrics = {"p50": "p50", 
+                    "p90": "p90", 
+                    "p99": "p99", 
+                    "mem": "mem_MB_max_fortioserver_deployment_proxy", 
+                    "cpu": "cpu_mili_max_fortioserver_deployment_proxy"}
 
-    if metric is None:
+    if y_label_short is None:
         sys.exit('need metric')
-    if metric not in valid_metrics: 
+    if y_label_short not in valid_metrics: 
         sys.exit("invalid metric")
     if csv is None:
         sys.exit('need CSV file') 
     
-    m = valid_metrics[metric] #the CSV label
+    y_label = valid_metrics[y_label_short] #the CSV label
 
     # 1. read CSV to pandas dataframe 
     df = pd.read_csv(csv, index_col=None, header=0)
     df["Labels"] = [ x.split('_', 6)[-1] for x in df['Labels']]
 
-    # 2. generate series to plot (x=connections (numthreads), y=metric) 
-    c, series = get_series(df, m) 
+    # 2. generate series to plot (x= numthreads or qps, y=y_) 
+    x_series, y_series = get_series(df, x_label, y_label) 
 
     # 3. generate title 
     qps=df.at[0, 'ActualQPS'] 
-    seconds=df.at[0, 'ActualDuration']
-    title="Istio {}, {} QPS over {}s".format(metric, qps, seconds)
+    seconds=df.at[0, 'ActualDuration']  
+    threads=df.at[0, 'NumThreads']  
+
+    if x_label == "connections": #option 1 -- constant QPS, numthreads x metric 
+        title="{} {}, {} QPS over {} seconds".format(mesh, y_label_short, qps, seconds)
+    else: # option 2 -- constant numthreads, QPS x metric 
+        title="{} {}, {} threads over {} seconds".format(mesh, y_label_short, threads, seconds)
+
 
     # 4. prep file-write 
     fn = "".join(title.split())
     f = "/tmp/" + fn + ".html" 
     output_file(f) 
 
     # 5. create chart -> save as interactive HTML 
-    p = build_chart(title, metric, c, series)  
+    p = build_chart(title, x_label, x_series, y_label, y_series)
```

Comment:
While you are here can you fix the file extensions of the output?
They should be .csv and .json I don’t think that is happening.